syntax = "proto3";


package executor;

service Executor {
    rpc GetValue(ValueRequest) returns(ValueResponse) {}
    rpc AddValueToBuffer(RemoteValue) returns(BufferResponse) {}
    rpc RunComputation(ComputeRequest) returns(ComputeResponse) {}
}

// TODO Reduce the number of messages
// Should be a message holding a stream of bytes
// handlers would be responsible for reconstructing 
// object based on stream of bytes
message RemoteValue {
    float value = 1;
    string rendezvous_key = 2;
    int64 session_id = 3;
}

message ValueRequest {
    string rendezvous_key = 1;
    int64 session_id = 2;
}

message ValueResponse {
    float value = 1;
}

message BufferResponse {}

message ComputeRequest {
    bytes computation = 1;
    string role = 2;
    int64 session_id = 3;
}

message ComputeResponse {}
